metadata/src/test/java/org/apache/kafka/image/writer/ImageWriterOptionsTest.java

```diff
@@ -48,40 +48,33 @@
 public class ImageWriterOptionsTest {
     @Test
     public void testDefaultLossHandler() {
-        ImageWriterOptions options = new ImageWriterOptions.Builder().build();
+        ImageWriterOptions options = new ImageWriterOptions.Builder(MetadataVersion.latestProduction()).build();
         assertEquals("stuff", assertThrows(UnwritableMetadataException.class,
                 () -> options.handleLoss("stuff")).loss());
     }
 
     @Test
     public void testSetMetadataVersion() {
```

Comment:
If we rewrite `ImageWriterOptions` by record class, this `testSetMetadataVersion` is unnecessary as all are pure setters.

Reply:
I agree that this test isn't useful since the special logic for the setter was removed - even without removing the builder.